kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/NtpdClockSyncProvider.java

```diff
@@ -35,14 +40,14 @@
     @Override
     protected boolean syncClock() throws KuraException {
         boolean ret = false;
-        SafeProcess proc = null;
         try {
             // Execute a native Linux command to perform the NTP time sync.
             int ntpTimeout = this.ntpTimeout / 1000;
-            proc = exec("ntpdate -t " + ntpTimeout + " " + this.ntpHost);
-            proc.waitFor();
-            if (proc.exitValue() == 0) {
-                logger.info("System Clock Synchronized with " + this.ntpHost);
+            Command command = new Command("ntpdate -t " + ntpTimeout + " " + this.ntpHost);
+            command.setTimeout(60);
+            CommandStatus status = this.executorService.execute(command);
+            if ((Integer) status.getExitStatus().getExitValue() == 0) {
```

Comment:
why do we need to cast it to Integer?

Reply:
Because the getExitValue() returns a generic Object.